### PR TITLE
parser: allow whitespace control on `extends`

### PR DESCRIPTION
**Pull request by Kijewski**
Using whitespace control on `extends`, e.g. `{%- extends "some.html" ~%}`, is an explicit syntax error. The test was added in <https://github.com/rinja-rs/rinja/commit/29687db996307463ac2f4de01f67704e909dd1fe>.

IMHO, the syntax should be allowed, even though it does not change anything if you add whitespace control, because anything outside of `{% block %}`s gets omitted. Whitespace control on the far sides of `{% block %}`s is not a syntax error, either. For consistency, I like to always write `{%- -%}` on the outmost scope, regardless if I have to or not.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Can you add a test to check the generated result please? (Should not have any impact but better to ensure it with a test)

**Comment by Kijewski** [comment]
I added an exhaustive test with all combinations of control characters, but I'm not 100% sure that is what you meant. :-/

**Comment by GuillaumeGomez** [comment]
It's not but it's a good add already. :wink:  What I meant is checking how whitespace suppressors are handled when you `render()` the template.

**Comment by Kijewski** [comment]
Anything outside of `{% block %}` is discarded in extended templates, anyway. For the generator nothing has hanged.

**Comment by GuillaumeGomez** [comment]
And that's exactly what we need to check (if we didn't already).

**Comment by Kijewski** [comment]
I added a test that the same source code is generated, too.

File: rinja_derive/src/tests.rs [hunk 1]
```diff
@@ -135,6 +112,34 @@ struct Foo {{ {} }}"##,
     }
 }
 
+fn jinja_to_rust(jinja: &str, fields: &[(&str, &str)]) -> syn::Result<syn::File> {
```
Comment (GuillaumeGomez) [inline rinja_derive/src/tests.rs:115]: No no, I meant like https://github.com/rinja-rs/rinja/blob/master/testing/tests/simple.rs#L13-L35. Sorry, should have been more clear. No need to check the generated rust code here.

Comment (Kijewski) [reply rinja_derive/src/tests.rs:115]: If the same code is generated, then the same code is rendered. It's simpler this way than writing tests with 4 + 4 = 8 cases (all variants at the front and back), or 4 * 4 = 16 tests (cross of all variants).

Comment (GuillaumeGomez) [reply rinja_derive/src/tests.rs:115]: I'm not convinced but there are tests, so all good for me.

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**